src/main/java/lambdaperson/FilteredPersons.java

```diff
@@ -0,0 +1,23 @@
+package lambdaperson;
+
+import personlist.Person;
+import java.util.Comparator;
+import java.util.LinkedHashSet;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class FilteredPersons {
+
+    public static Set<Person> filterPersons(Set<Person> personList) {
+        Comparator<Person> sortingByNameLength =
+                Comparator.comparingInt((Person p) -> p.getName().length());
+
+        return personList.stream().filter(Person -> Person.getAge() < 30 & Person.getAge() > 20)
```

Comment:
`Person` - it's a variable inside lambda, and Java code style recommends to name variables in camelCase, so it's better to name it `person`.
BTW, sometimes it's convenient to give such variables quite short names, such as: `o`, `it`, etc. It's up to author to choose a better one for readability